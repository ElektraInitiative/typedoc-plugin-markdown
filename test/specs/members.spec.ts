import * as Handlebars from 'handlebars';
import { ProjectReflection } from 'typedoc';

describe(`Members:`, () => {
  let project: ProjectReflection;
  let membersPartial: Handlebars.TemplateDelegate;
  let memberPartial: Handlebars.TemplateDelegate;
  beforeAll(async () => {
    project = await global.bootstrap(['members.ts']);

    global.stubPartials(['member', 'index', 'member.sources']);
    global.stubHelpers(['relativeURL']);
    membersPartial = global.getPartial('members');
    memberPartial = global.getPartial('member');
  });

  describe(`(members)`, () => {
    test(`should compile module members'`, () => {
      expect(
        global.compileTemplate(
          membersPartial,
          global.findModule(project, 'members'),
        ),
      ).toMatchSnapshot();
    });

    test(`should compile class members'`, () => {
      expect(
        global.compileTemplate(
          membersPartial,
          project.getChildByName('ClassWithAccessorMembers'),
        ),
      ).toMatchSnapshot();
    });
  });

  describe(`(member)`, () => {
    test(`should compile declaration members'`, () => {
      expect(
        global.compileTemplate(
          memberPartial,
          project.getChildByName('declarationMember'),
        ),
      ).toMatchSnapshot();
    });

    test(`should compile a signature members'`, () => {
      expect(
        global.compileTemplate(
          memberPartial,
          project.getChildByName('signatureMember'),
        ),
      ).toMatchSnapshot();
    });

    test(`should compile members with getter'`, () => {
      expect(
        global.compileTemplate(
          memberPartial,
          (
            project.getChildByName('ClassWithAccessorMembers') as any
          ).getChildByName('getter'),
        ),
      ).toMatchSnapshot();
    });

    test(`should compile members with setter'`, () => {
      expect(
        global.compileTemplate(
          memberPartial,
          (
            project.getChildByName('ClassWithAccessorMembers') as any
          ).getChildByName('setter'),
        ),
      ).toMatchSnapshot();
    });
  });

  describe(`(with hideMembersSymbol)`, () => {
    let project: ProjectReflection;
    beforeAll(async () => {
      project = await global.bootstrap(['members.ts'], {
        hideMembersSymbol: true,
      });
      global.stubPartials(['member', 'member.sources']);

      memberPartial = global.getPartial('member');
    });

    test(`should compile members without special symbols`, () => {
      expect(
        global.compileTemplate(
          memberPartial,
          project.getChildByName('signatureMember'),
        ),
      ).toMatchSnapshot();
    });
  });
});
