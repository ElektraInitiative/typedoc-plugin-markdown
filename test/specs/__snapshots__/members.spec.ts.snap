// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Members: (member) should compile a signature members' 1`] = `
"### signatureMember

▸ **signatureMember**(): \`void\`

#### Returns

\`void\`

[partial: member.sources]

___
"
`;

exports[`Members: (member) should compile declaration members' 1`] = `
"### declarationMember

• **declarationMember**: \`string\`

[partial: member.sources]

___
"
`;

exports[`Members: (member) should compile members with getter' 1`] = `
"### getter

• \`get\` **getter**(): \`string\`

#### Returns

\`string\`

[partial: member.sources]

___
"
`;

exports[`Members: (member) should compile members with setter' 1`] = `
"### setter

• \`set\` **setter**(\`value\`): \`void\`

#### Parameters

| Name | Type |
| :------ | :------ |
| \`value\` | \`string\` |

#### Returns

\`void\`

[partial: member.sources]

___
"
`;

exports[`Members: (members) should compile class members' 1`] = `
"## Constructors

[partial: member]

## Properties

[partial: member]

## Accessors

[partial: member]

[partial: member]
"
`;

exports[`Members: (members) should compile module members' 1`] = `
"
"
`;

exports[`Members: (with hideMembersSymbol) should compile members without special symbols 1`] = `
"### signatureMember

**signatureMember**(): \`void\`

#### Returns

\`void\`

[partial: member.sources]

___
"
`;
